requests/models.py

```diff
@@ -154,8 +154,12 @@ def _encode_files(files, data):
 
             if isinstance(fp, (str, bytes, bytearray)):
                 fdata = fp
-            else:
+            elif hasattr(fp, 'read'):
```

Comment:
@ueg1990 that's because properties in python2 aren't attributes. very known and undertstood functionality.